drivers/d3d12/rendering_context_driver_d3d12.cpp

```diff
@@ -43,12 +43,20 @@
 #pragma GCC diagnostic ignored "-Wshadow"
 #pragma GCC diagnostic ignored "-Wswitch"
 #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#elif defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnon-virtual-dtor"
+#pragma clang diagnostic ignored "-Wstring-plus-int"
+#pragma clang diagnostic ignored "-Wswitch"
+#pragma clang diagnostic ignored "-Wmissing-field-initializers"
```

Comment:
I believe Clang honors `#pragma GCC diagnostic` too, so as long as the warnings to ignore are the same, we could maybe just remove the `!defined(__clang__)` part in the first block and avoid the redundancy.

Reply:
Well I see there's already a difference with `-Wshadow` on GCC and `-Wstring-plus-int` on Clang, so maybe not worth it.

Reply:
Another option for Clang would be to ignore all warnings:
```cpp
#pragma clang diagnostic ignored "-Wall"
#pragma clang diagnostic ignored "-Wextra"
```
[GCC doesn't seem to support that](https://stackoverflow.com/questions/32049296/how-to-disable-all-warnings-using-pragma-directives-in-gcc) so it still needs each specific warning listed manually.

Reply:
Since it is for DX headers we might as well suppress everything, I'll check if it works like this a bit later.

Reply:
> Another option for Clang would be to ignore all warnings

It doesn't work (at least not for all warnings).